pkg/snmp/snmp.go

```diff
@@ -107,6 +123,27 @@ func NewListenerConfig() (ListenerConfig, error) {
 		if config.Loader == "" {
 			config.Loader = snmpConfig.Loader
 		}
+		if config.Community == "" && config.CommunityStringLegacy != "" {
```

Comment:
🤏 could be refactored in something like
`config.Community = useLegacyIfEmpty(config.Community, config.CommunityLegacy)`

Reply:
@pducolin good idea. Fixed here: https://github.com/DataDog/datadog-agent/pull/8227/commits/0e3391f95bc20adb0e4c27d0110e30a610f9627f

Related: https://github.com/golang/go/issues/14423